server/src/test/java/org/apache/druid/server/coordinator/duty/KillUnusedSegmentsTest.java

```diff
@@ -96,362 +104,805 @@ public class KillUnusedSegmentsTest
+            dayOldSegment.getInterval().getStart(),
+            hourOldSegment.getInterval().getEnd()
+        )
+    );
 
-    mockTaskSlotUsage(1.0, Integer.MAX_VALUE, 1, 10);
-    runAndVerifyKillInterval(new Interval(
-        dayOldSegment.getInterval().getStart(),
-        hourOldSegment.getInterval().getEnd()
-    ));
-    verifyState(ImmutableMap.of(DATASOURCE, hourOldSegment.getInterval().getEnd()));
+    log.info("Third run...");
+    final DruidCoordinatorRuntimeParams thirdRun = killDuty.run(paramsBuilder.build());
 
-    mockTaskSlotUsage(1.0, Integer.MAX_VALUE, 1, 10);
-    runAndVerifyKillInterval(new Interval(
-        nextDaySegment.getInterval().getStart(),
-        nextMonthSegment.getInterval().getEnd()
-    ));
-    verifyState(ImmutableMap.of(DATASOURCE, nextMonthSegment.getInterval().getEnd()));
-    verifyStats(9, 1, 10, 3);
+    validateStats(
+        new ExpectedStats(30, 3, 30, ImmutableMap.of(DS1, 6L)),
+        thirdRun.getCoordinatorStats()
+    );
+
+    validateAndResetState(
+        DS1,
+        new Interval(
+            nextDaySegment.getInterval().getStart(),
+            nextMonthSegment.getInterval().getEnd()
+        )
+    );
   }
 
   @Test
   public void testKillTaskSlotRatioNoAvailableTaskCapacityForKill()
   {
-    mockTaskSlotUsage(0.10, 10, 1, 5);
-    runAndVerifyNoKill();
-    verifyState(ImmutableMap.of());
-    verifyStats(0, 0, 0);
+    dynamicConfigBuilder.withKillTaskSlotRatio(0.10);
+    dynamicConfigBuilder.withMaxKillTaskSlots(10);
+    paramsBuilder.withDynamicConfigs(dynamicConfigBuilder.build());
+
+    overlordClient = new TestOverlordClient(1, 5);
+
+    overlordClient.addTask(DS1);
+
+    final KillUnusedSegments killDuty = new KillUnusedSegments(
+        segmentsMetadataManager,
+        overlordClient,
+        configBuilder.build()
+    );
+
+    final DruidCoordinatorRuntimeParams runParams = killDuty.run(paramsBuilder.build());
+
+    validateStats(
+        new ExpectedStats(0, 0, 0, ImmutableMap.of()),
+        runParams.getCoordinatorStats()
+    );
+    validateAndResetState(DS1, null);
   }
 
   @Test
   public void testMaxKillTaskSlotsNoAvailableTaskCapacityForKill()
   {
-    mockTaskSlotUsage(1.0, 3, 3, 10);
-    runAndVerifyNoKill();
-    verifyState(ImmutableMap.of());
-    verifyStats(0, 0, 3);
+    dynamicConfigBuilder.withKillTaskSlotRatio(1.0);
+    dynamicConfigBuilder.withMaxKillTaskSlots(3);
+    paramsBuilder.withDynamicConfigs(dynamicConfigBuilder.build());
+
+    overlordClient = new TestOverlordClient(3, 10);
+    overlordClient.addTask(DS1);
+    overlordClient.addTask(DS1);
+    overlordClient.addTask(DS1);
+
+    final KillUnusedSegments killDuty = new KillUnusedSegments(
+        segmentsMetadataManager,
+        overlordClient,
+        configBuilder.build()
+    );
+
+    final DruidCoordinatorRuntimeParams runParams = killDuty.run(paramsBuilder.build());
+
+    validateStats(
+        new ExpectedStats(0, 0, 3, ImmutableMap.of()),
```

Comment:
I feel it would be much more readable to just validate the relevant stats individually, rather than having to pass zeroes and empty maps repeatedly. This would allow someone reading the test to instantly recognize exactly what is being verified.

You can see some examples in `RunRulesTest`.
Something like:
```
Assert.assertEquals(10L, stats.getSegmentStat(Stats.Segments.ASSIGNED, "normal", DATASOURCE));
```

Reply:
Yes, definitely, it reads more clearly now!